lib/provider.js

```diff
@@ -335,8 +334,9 @@ class StorageProvider {
       );
     }
     const { Execute: execute } = foundAction.definition;
+    const { context, session } = actionArgs;
     const stripFn = this.loggingEnabled ? x => x : LogStatus.strip;
-    return execute(this, ...args).then(stripFn, err => {
+    return execute(this, context, session, args).then(stripFn, err => {
```

Comment:
I'd prefer to have this signature as `execute(this, session, context, args)` or maybe even put `session` in the `context` (`context.session`) to avoid having to specify it when it may not be needed.